presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -185,26 +183,27 @@ private static String hiveTableProperties(TransactionalTableType transactionalTa
 
     private static void compactTableAndWait(CompactionMode compactMode, String tableName, String partitionString, Duration timeout)
     {
-        Set<String> existingCompactionsIds = getTableCompactions(compactMode, tableName)
-                .map(row -> row.get("compactionid"))
-                .collect(toUnmodifiableSet());
-
+        assertEquals(getTableCompactions(compactMode, tableName).count(), 0);
         onHive().executeQuery(format("ALTER TABLE %s %s COMPACT '%s'", tableName, partitionString, compactMode.name()));
 
-        Set<String> tableCompactionsIds = Sets.difference(getTableCompactions(compactMode, tableName)
-                .map(row -> row.get("compactionid"))
-                .collect(toUnmodifiableSet()),
-                existingCompactionsIds);
-
-        assertFalse(tableCompactionsIds.isEmpty(), "tableCompactionsIds are empty");
-
+        // Since we disabled table auto compaction and we checked that there are no compaction to the table
+        // we can assume that every compaction from now on is triggered in this test
+        // and all compaction should complete successfully before proceeding.
         assertEventually(timeout, () -> {
-            Set<String> completedCompactionsIds = getTableCompactions(compactMode, tableName)
+            List<Map<String, String>> allCompactions = getTableCompactions(compactMode, tableName)
```

Comment:
shouldn't there be just 1 row with state == `SUCCEEDED`.

And then we can just do:
```
        Map<String, String> compaction = getOnlyElement(
                getTableCompactions(compactMode, tableName).collect(ImmutableList.toImmutableList()));
        assertEquals(compaction.get("state"), "succeeded");

```

Reply:
Yes, there should be only one row with either state succeeded or failed